src/frequenz/client/base/grpc_streaming_helper/_grpc_streaming_helper.py

```diff
@@ -0,0 +1,108 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Implementation of the grpc streaming helper."""
+
+import asyncio
+import logging
+import typing
+
+import grpc
+from grpc.aio import UnaryStreamCall  # type: ignore[import]
+
+from frequenz import channels
+
+from .. import retry_strategy
+
+_logger = logging.getLogger(__name__)
+
+
+_InputT = typing.TypeVar("_InputT")
+_OutputT = typing.TypeVar("_OutputT")
+
+
+class GrpcStreamingHelper(typing.Generic[_InputT, _OutputT]):
+    """Helper class to handle grpc streaming methods."""
+
+    def __init__(
+        self,
+        stream_name: str,
+        stream_method: typing.Callable[[], UnaryStreamCall[typing.Any, _InputT]],
+        transform: typing.Callable[[_InputT], _OutputT],
+        retry_spec: retry_strategy.RetryStrategy | None = None,
+    ):
+        """Initialize the streaming helper.
+
+        Args:
+            stream_name: A name to identify the stream in the logs.
+            stream_method: A function that returns the grpc stream. This function is
+                called everytime the connection is lost and we want to retry.
+            transform: A function to transform the input type to the output type.
+            retry_spec: The retry strategy to use, when the connection is lost. Defaults
+                to retries every 3 seconds, with a jitter of 1 second, indefinitely.
+        """
+        self._stream_name = stream_name
+        self._stream_method = stream_method
+        self._transform = transform
+        self._retry_spec = (
+            retry_strategy.LinearBackoff() if retry_spec is None else retry_spec.copy()
```

Comment:
I wrote a comment to suggest to make retry specs/strategy immutable, so we don't need to copy them. But now I see retry strategies are actually keeping state, so it is not really a spec or a strategy, is more like a tracker, that you need to reset when a connection is complete, so the copy makes sense, but it might be nice to look for a clearer name in the future (in a different PR), `RetryTracker` or something more along those lines.